### PR TITLE
feat: add gatlingEnvPlugin CLOUD-1511

### PR DESCRIPTION
**Pull request by bastien-gatling**
Motivation:
Some config will provided by environment variable. This plugin allow to set files with env var that will be present in the fork process

Modifications:
Add gatlingEnvPlugin

Result:
A new plugin.

Ref: CLOUD-1511

### CODE REVIEW

File: src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+   * @return all env value as a map
+   */
+  def readEnvFile(file: File): Map[String, String] = {
+    Using(Source.fromFile(file)) { source =>
+      LINE_REGEX.findAllMatchIn(source.mkString).map(kv => (kv.group(1), unescapeCharacters(kv.group(2)))).toMap
+    }.getOrElse(Map.empty)
+  }
+
+  private def unescapeCharacters(value: String): String = {
+    value.replaceAll("""\\([^$])""", "$1")
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala:56]: Why not `$`?

```
FOO=\$BAR
```

File: src/main/scala/io/gatling/build/environment/GatlingEnvPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  object GatlingEnvPluginKeys extends GatlingEnvPluginKeys
+
+  import autoImport._
+
+  override lazy val globalSettings: Seq[Def.Setting[_]] = Seq(
+    gatlingEnvFiles := Seq.empty
+  )
+
+  override lazy val projectSettings: Seq[Def.Setting[_]] = Seq(
+    gatlingEnvVars / fileInputs ++= gatlingEnvFiles.value.map(file => Glob(file)),
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/environment/GatlingEnvPlugin.scala:46]:
```suggestion
    gatlingEnvVars / fileInputs ++= gatlingEnvFiles.value.map(Glob.apply),
```

File: src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala (hunk 1 above)
Comment (bastien-gatling) [reply src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala]: Yep, you're right.

Actually, as we doesn't handle `'` or `"`, unescaping seems not useful for our usage, so i remove this method.

File: src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  private val LINE_REGEX =
+    """(?xms)
+       (?:^|\A)           # start of line
+       (                  # start variable name (captured)
+         [a-zA-Z_]        # single alphabetic or underscore character
+         [a-zA-Z0-9_.-]*  # zero or more alphnumeric, underscore, period or hyphen
+       )                  # end variable name (captured)
+       (?:=)              # assignment
+       (                  # start variable value (captured)
+         [^\#\r\n]*       # variable
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala:35]: As we do not manage comment on line anymore and we don't manage escape, we shouldn't forbid shard `#` in the value either!

What I have in mind: url with fragment

```
REDIRECT_URL=https://auth.cloud.gatling.io/login#register
```

Comment (bastien-gatling) [reply src/main/scala/io/gatling/build/environment/EnvironmentUtils.scala]: Good catch !

I've added a test case.

**Review by Isammoc — APPROVED** [review]

**Review by guilgaly — APPROVED** [review]

**Pull request merged**